### PR TITLE
Allow users to specify total_count_max in Multinomial

### LINKED ISSUE #1556: total count as model parameter in multinomial distribution
Here is the issue corresponding to the forum question [5066](https://forum.pyro.ai/t/multinomial-plate-and-total-counts/5066).
Example:
5 Customers are going to buy apples from 4 available category of apples.
- The total number of purchase is negative binomial (with zero purchase being the most common one).
- The probability of choosing apples from each category is drawn from Dirichlet distribution for each customer.
Here is the minimal code:
```python
import jax
import jax.numpy as jnp
import matplotlib.pyplot as plt
import numpyro
import numpyro.distributions as dist
from numpyro.infer import Predictive

seed = 1234
# numpyro seed
rng_key = jax.random.PRNGKey(seed)
rng_trace, rng_prior_pred, rng_posterior_pred = jax.random.split(rng_key, 3)

def gen_data(num_apple_category: int = 4, n_customers: int = 5):
    mu = 1
    alpha = 2
    with numpyro.plate("customers", n_customers):
        # The latent probability for number of buys for each customer is Negative binomial.
        # The lower number of the purchase has more chance.
        num_purchase = numpyro.sample("num_purchase", dist.NegativeBinomial2(mu, alpha))
        desired_probs = numpyro.sample(
            "desired_probs", dist.Dirichlet(0.5 * jnp.ones(num_apple_category))
        )
        numpyro.sample(
            "apples", dist.Multinomial(total_count=num_purchase, probs=desired_probs)
        )


num_samples = 1
# Draw from priors to make the fake data
prior_predictive = Predictive(gen_data, num_samples=num_samples)
prior_predictions = prior_predictive(rng_prior_pred)
```

Error:
```bash
AssertionError: The total count parameter `n` should not be a jax abstract array.
```

Versions:
```bash
jax=='0.4.1'
jaxlib=='0.4.1'
numpyro=='0.10.1'
```

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #1556

### CODE REVIEW

**Review by martinjankowiak — APPROVED** [review]
lgtm but i wonder if these multinomial distributions need docstrings?

File: numpyro/distributions/discrete.py [hunk 1]
```diff
@@ -618,11 +632,31 @@ def infer_shapes(logits, total_count):
...
 
 
-def Multinomial(total_count=1, probs=None, logits=None, *, validate_args=None):
+def Multinomial(
+    total_count=1, probs=None, logits=None, *, total_count_max=None, validate_args=None
+):
+    """Multinomial distribution.
+
+    :param total_count: number of trials. If this is a JAX array,
+        `total_count_max` is required to specify.
```
Comment (martinjankowiak) [inline numpyro/distributions/discrete.py:641]: => it is required to specify `total_count_max`.

File: numpyro/distributions/discrete.py [hunk 2]
```diff
@@ -618,11 +632,31 @@ def infer_shapes(logits, total_count):
...
+    :param probs: event probabilities
+    :param logits: event log probabilities
+    :param int total_count_max: the maximum number of trials
```
Comment (martinjankowiak) [inline numpyro/distributions/discrete.py:644]: something like this? => the maximum number of trials, i.e. `max(total_count)`.

**Review by martinjankowiak — APPROVED** [review]

File: numpyro/distributions/discrete.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/distributions/discrete.py]: done

**Review by martinjankowiak — DISMISSED** [review]

**Comment by martinjankowiak** [comment]
some tests are failing are these just flaky?

> test/infer/test_mcmc.py::test_beta_bernoulli_x64

**Review by martinjankowiak — APPROVED** [review]

**Pull request merged**